nginx_module/src/http_request.rs

```diff
@@ -57,6 +60,59 @@ impl<'a, Ctx: Default> HttpRequestAndContext<'a, Ctx> {
             Ok((req, ctx))
         }
     }
+
+    // get raw pointer to request structure
+    pub fn get_request_pointer (
+        &mut self,
+    ) -> anyhow::Result<*mut ngx_http_request_t> {
+            let req = &mut self.0 as *mut ngx_http_request_t;
+            if req.is_null() {
+                anyhow::bail!("tried to get request pointer which is null");
+            }
+            Ok(req)
+    }
+
+    // send prepared headers
+    // headers should be prepared in headers_out structure
+    pub fn send_headers(&mut self) -> anyhow::Result<()> {
+        unsafe {
+            if 0==ngx_http_send_header(&(self.0) as *const ngx_http_request_t as *mut ngx_http_request_t) {
```

Comment:
Please use NGX_OK instead of 0 here, to clearly state what the result code means.

Reply:
fixed